qiskit/transpiler/preset_passmanagers/level1.py

```diff
@@ -105,14 +109,63 @@ def level_1_pass_manager(pass_manager_config: PassManagerConfig) -> PassManager:
     # 1. Use trivial layout if no layout given
     _given_layout = SetLayout(initial_layout)
 
-    _choose_layout_and_score = [
-        TrivialLayout(coupling_map),
-        Layout2qDistance(coupling_map, property_name="trivial_layout_score"),
-    ]
-
     def _choose_layout_condition(property_set):
         return not property_set["layout"]
 
+    def _trivial_not_perfect(property_set):
+        # Verify that a trivial layout is perfect. If trivial_layout_score > 0
+        # the layout is not perfect. The layout is unconditionally set by trivial
+        # layout so we need to clear it before contuing.
+        if property_set["trivial_layout_score"] is not None:
+            if property_set["trivial_layout_score"] != 0:
+                return True
+            else:
+                warnings.warn(
+                    "The current implicit default of using a trivial layout if it's a "
+                    "perfect layout will change in a future release, if you're "
+                    "depending on this behavior it is better to explicit set "
+                    "layout_method='trivial' when calling transpile()",
+                    FutureWarning,
+                )
+                # stack_level is not set because targeting transpile() is difficult
+        return False
+
+    def _vf2_match_not_found(property_set):
+        # If a layout hasn't been set by the time we run vf2 layout we need to
+        # run layout
+        if property_set["layout"] is None:
+            return True
+        # if VF2 layout stopped for any reason other than solution found we need
+        # to run layout since VF2 didn't converge.
+        if (
+            property_set["VF2Layout_stop_reason"] is not None
+            and property_set["VF2Layout_stop_reason"] is not VF2LayoutStopReason.SOLUTION_FOUND
+        ):
+            return True
+        return False
+
+    _choose_layout_0 = (
+        []
+        if pass_manager_config.layout_method
+        else [
+            TrivialLayout(coupling_map),
+            Layout2qDistance(coupling_map, property_name="trivial_layout_score"),
+        ]
+    )
+
+    _choose_layout_1 = (
+        []
+        if pass_manager_config.layout_method
+        else VF2Layout(
+            coupling_map,
+            seed=seed_transpiler,
+            call_limit=int(5e4),  # Set call limit to ~100ms with retworkx 0.10.2
```

Comment:
Does the comment imply some action is needed now that retworkx 0.10.2 and 0.11 are out?  Or is just saying that `50_000` _is_ ~100ms on retworkx 0.10.2?

Reply:
Yeah, it's to indicate we're using 50_000 as a proxy for roughly 100ms (with retworkx 0.10.2). I was just trying to document where the magic number comes from and why it's set to 50k.